### PR TITLE
feat: actions for connections pool feature

### PR DESCRIPTION
**Pull request by diegoreis42**
This PR propose a new feature rather than addressing any specific issue, it uses the `Context Menu` component to add update and delete capabilities to users. 

It's triggered by right-clicking on the desired row, I thought about using the "three dot" icon but this approach seemed more flexible and let's the door open to others actions. But if you'd prefer the icon-based method I'd be glad to make that change. 

## How it looks?

Screencast from 2024-09-30 17-01-36

### CODE REVIEW

File: src/app/(main)/connections/_components/modal.tsx [hunk 1]
```diff
@@ -13,16 +11,14 @@ import {
...
 import { z } from "zod";
 
 const formSchema = z.object({
   name: z.string().min(1, { message: "Name is required." }),
-  host: z
-    .string()
-    .min(1, { message: "Host is required." })
-    .regex(/^[a-zA-Z0-9.-]+$/, "Invalid host format."),
+  host: z.string().ip(),
   username: z.string().min(1, { message: "Username is required." }),
```
Comment (DanielHe4rt) [inline src/app/(main)/connections/_components/modal.tsx:22]: Since we're already going through this feature:

usernames usually follows the idea of `no spacing`, so this can be a rule to implement.

File: src/app/(main)/connections/_components/modal.tsx [hunk 2]
```diff
@@ -13,16 +11,14 @@ import {
...
   password: z.string().min(1, { message: "Password is required." }),
   dc: z.string().min(1, { message: "Data center (DC) is required." }),
```
Comment (DanielHe4rt) [inline src/app/(main)/connections/_components/modal.tsx:24]: I forgot to mention, but not all items on that table is actually a "fillable". @Daniel-Boll please correct me, but we can retrieve the DC structure directly from the Driver. Correct? This will enable us to know by a single connection the topology of the entire cluster after the first connection.

File: src/app/(main)/connections/_components/modal.tsx [hunk 3]
```diff
@@ -13,16 +11,14 @@ import {
...
 import { FormProvider, useForm } from "react-hook-form";
 import { z } from "zod";
 
 const formSchema = z.object({
   name: z.string().min(1, { message: "Name is required." }),
-  host: z
-    .string()
-    .min(1, { message: "Host is required." })
-    .regex(/^[a-zA-Z0-9.-]+$/, "Invalid host format."),
+  host: z.string().ip(),
```
Comment (DanielHe4rt) [inline src/app/(main)/connections/_components/modal.tsx:21]: Make it required.

**Review by DanielHe4rt — CHANGES_REQUESTED** [review]
Here's some business rules to cover, after it seems ready to merge IMO.

File: src/app/(main)/connections/_components/modal.tsx (hunk 2 above)
Comment (Daniel-Boll) [reply src/app/(main)/connections/_components/modal.tsx:24]: Ngl, I don't remember, never used the DC structure query, but we may be able to get it from the driver.

**Comment by Daniel-Boll** [comment]
Since we are already on this topic, I forgot to mention in the last PR, but this solution is local only because the way we are managing the SQLite connections, if we plan to host anything at `scylla.studio` without the user having to ran it, then this solution will have to be adapted.

No need to change this now, just so we get in the same page @Daniel-Boll @diegoreis42 :wave:

**Comment by diegoreis42** [comment]
@Daniel-Boll sure! I tried to add some abstractions layers between the components and the db management to avoid too much headache in such scenarios. But ya, in the current state without managing the user session on queries nor making security stuff like encryption it is not suitable to use in a SaaS manner. 

Actually would be nice to talk about the project MVP goals. I'm looking forward to help.

File: src/app/(main)/connections/_components/connection-table.tsx [hunk 4]
```diff
@@ -89,13 +89,17 @@ export default function ConnectionTableServer() {
...
+                {["name", "host", "username", "password", "dc", "nodes"].map(
                   (key) => (
                     <ContextMenu key={`${conn.name}-${key}`}>
                       <ContextMenuTrigger asChild>
                         <TableCell>
-                          {key === "password" ? "••••••••" : (conn as any)[key]}
-                        </TableCell>
+                          {key === "password"
+                            ? "••••••••"
+                            : key === "dc" // since dc will be retrived from the driver it is hardcoded for now
```
Comment (DanielHe4rt) [inline src/app/(main)/connections/_components/connection-table.tsx:99]: Instead hardcoded it here, you can just add directly into the INSERT query. Furthermore we would be tweaking there to get these missing numbers.

Any Database with Partition Tolerance would have 1 or more nodes, so you can just expect it.

TLDR: remove this rule and add the `3` hardcdoed to the database.

File: src/app/(main)/connections/_components/modal.tsx [hunk 5]
```diff
@@ -13,38 +11,36 @@ import {
...
+  .object({
+    name: z
+      .string()
+      .trim()
+      .min(1, { message: "Name is required." })
+      .refine((value) => !/\s/.test(value), {
+        message: "Name cannot contain spaces.",
+      }),
+    host: z.string().ip(),
+    username: z.string().min(1, { message: "Username is required." }),
```
Comment (DanielHe4rt) [inline src/app/(main)/connections/_components/modal.tsx:28]: Btw, the "username" was the field that you should trim, not the name.
```suggestion
    username: z.string().trim().min(1, { message: "Username is required." }),
```

File: src/app/(main)/connections/_components/modal.tsx (hunk 2 above)
Comment (DanielHe4rt) [reply src/app/(main)/connections/_components/modal.tsx]: We can check it tomorrow then.

**Review by DanielHe4rt — APPROVED** [review]
LGTM!

File: .gitignore [hunk 6]
```diff
@@ -34,3 +34,6 @@ yarn-error.log*
 # typescript
 *.tsbuildinfo
 next-env.d.ts
+
+# SQLite files
+*.db
```
Comment (DanielHe4rt) [inline .gitignore:39]: please create a base db before update the .gitignore

**Review by DanielHe4rt — COMMENTED** [review]
Obrigado pela contribuição! Só tive um pequeno pedido de ajuste pra que a PR seja aceita. 

Caso você tenha interesse em contribuir em outras funcionalidades, me avise para que eu possa abrir novas issues!
